src/nfv_test_api/v2/services/gnodeb.py

```diff
@@ -0,0 +1,211 @@
+"""
+       Copyright 2023 Inmanta
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+"""
+import logging
+import yaml
+import pathlib
+from typing import Any, Dict, List, Optional, Set, Union
+import subprocess
+
+import pydantic
+from pydantic import ValidationError
+from werkzeug.exceptions import Conflict, NotFound, ServiceUnavailable  # type: ignore
+
+from nfv_test_api.host import Host
+from nfv_test_api.v2.data.common import CommandStatus
+from nfv_test_api.v2.data.gnodeb import (
+    GNodeB,
+    GNodeBCreate,
+    GNodeBUpdate,
+)
+from nfv_test_api.v2.services.base_service import BaseService, K
+from nfv_test_api.config import Config
+
+LOGGER = logging.getLogger(__name__)
+
+class GNodeBServiceHandler:
+    def __init__(self) -> None:
+        self.processes: Dict[subprocess.Popen] = {}
+    
+    def add(self, identifier: str) -> None:
+        config_file_path = Config().gnb_config_folder+"gnb_"+identifier+".yml"
+        command = [
+            "nr-gnb",
+            "-c",
+            config_file_path
+        ]
+
+        out_file_path = Config().gnb_log_folder+"gnb_"+identifier+".log"
+        out = open(out_file_path,"w")
```

Comment:
In most cases you will actually create that file
```suggestion
        out = open(out_file_path,"w+")
```